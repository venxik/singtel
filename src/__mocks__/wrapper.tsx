import { Provider } from 'react-redux';
import { AppStore, RootState, setupStore } from '../redux';
import { PreloadedState } from '@reduxjs/toolkit';
import { PropsWithChildren } from 'react';
import { render } from '@testing-library/react-native';

interface ExtendedRenderOptions {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function AllTheProviders({ children }: { children: JSX.Element }) {
  return <Provider store={setupStore()}>{children}</Provider>;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
