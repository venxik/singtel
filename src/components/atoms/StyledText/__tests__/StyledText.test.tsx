import React from 'react';
import { renderWithProviders } from '../../../../__mocks__';
import ComicText from '../StyledText';

describe('StyledText', () => {
  it(`match snapshot`, () => {
    const tree = renderWithProviders(<ComicText>Snapshot test!</ComicText>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
