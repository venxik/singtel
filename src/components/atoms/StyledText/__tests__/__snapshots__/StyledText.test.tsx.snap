// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText match snapshot 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      [
        undefined,
        {
          "fontFamily": "comic",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
